vault/resource_nomad_secret_backend.go

```diff
@@ -0,0 +1,304 @@
+		Update: updateNomadAccessConfigResource,
+		Read:   readNomadAccessConfigResource,
+		Delete: deleteNomadAccessConfigResource,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: fields,
+	}
+}
+
+func createNomadAccessConfigResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	backend := d.Get("backend").(string)
+	description := d.Get("description").(string)
+	defaultTTL := d.Get("default_lease_ttl_seconds").(int)
+	local := d.Get("local").(bool)
+	maxTTL := d.Get("max_lease_ttl_seconds").(int)
+
+	log.Printf("[DEBUG] Mounting Nomad backend at %q", backend)
+	err := client.Sys().Mount(backend, &api.MountInput{
+		Type:        "nomad",
+		Description: description,
+		Local:       local,
+		Config: api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%ds", defaultTTL),
+			MaxLeaseTTL:     fmt.Sprintf("%ds", maxTTL),
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("error mounting to %q: %s", backend, err)
+	}
+
+	log.Printf("[DEBUG] Mounted Nomad backend at %q", backend)
+	d.SetId(backend)
+
+	data := map[string]interface{}{}
+	if v, ok := d.GetOkExists("address"); ok {
+		data["address"] = v
+	}
+
+	if v, ok := d.GetOkExists("ca_cert"); ok {
+		data["ca_cert"] = v
+	}
+
+	if v, ok := d.GetOkExists("client_cert"); ok {
+		data["client_cert"] = v
+	}
+
+	if v, ok := d.GetOkExists("client_key"); ok {
+		data["client_key"] = v
+	}
+
+	if v, ok := d.GetOkExists("max_token_name_length"); ok {
+		data["max_token_name_length"] = v
+	}
+
+	if v, ok := d.GetOkExists("token"); ok {
+		data["token"] = v
+	}
+
+	configPath := fmt.Sprintf("%s/config/access", backend)
+	log.Printf("[DEBUG] Writing %q", configPath)
+	if _, err := client.Logical().Write(configPath, data); err != nil {
+		return fmt.Errorf("error writing %q: %s", configPath, err)
+	}
+
+	log.Printf("[DEBUG] Wrote %q", configPath)
+	return readNomadAccessConfigResource(d, meta)
+}
+
+func readNomadAccessConfigResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+	log.Printf("[DEBUG] Reading %q", path)
+
+	mountResp, err := client.Sys().MountConfig(path)
+	if err != nil && util.Is404(err) {
+		log.Printf("[WARN] %q not found, removing from state", path)
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("error reading %q: %s", path, err)
+	}
+
+	d.Set("backend", d.Id())
+	d.Set("default_lease_ttl_seconds", mountResp.DefaultLeaseTTL)
+	d.Set("max_lease_ttl_seconds", mountResp.MaxLeaseTTL)
+
+	configPath := fmt.Sprintf("%s/config/access", d.Id())
+	log.Printf("[DEBUG] Reading %q", configPath)
+
+	resp, err := client.Logical().Read(configPath)
+	if err != nil {
+		return fmt.Errorf("error reading %q: %s", configPath, err)
+	}
+	log.Printf("[DEBUG] Read %q", configPath)
+	if resp == nil {
+		log.Printf("[WARN] %q not found, removing from state", configPath)
+		d.SetId("")
+		return nil
+	}
+
+	if val, ok := resp.Data["address"]; ok {
```

Comment:
I think these values should be read and update the state regardless if they are set in the response. If we use this resource to set them, and then out-of-band the backend is destroyed and recreated at the same address _but without these values_, we'll fail to detect the drift here.

Reply:
This is `Required: true`, so you would get an error if you didn't set it. Thoughts?